lib/blocks.php

```diff
@@ -108,6 +108,13 @@ function gutenberg_render_block( $block ) {
 		}
 	}
 
+	if ( isset( $block['innerBlocks'] )  && count( $block['innerBlocks'] ) ) {
+		$raw_content = $block['innerHTMLBeforeInnerBlocks'];
+		foreach ( $block['innerBlocks'] as $inner_block ) {
+			$raw_content .= gutenberg_render_block( $inner_block );
```

Comment:
although we shouldn't expect deeply nested blocks we may want to consider the possibility of their existence. suppose someone made a post with 1000 levels of nesting - that could theoretically cause WordPress to crash on account of overflowing the call stack.

I'd recommend considering one of two changes to the behavior here:
 - set a limit on the recursion and abort with some defined behavior when we're too deep
 - flip the recursion on its head and recurse using a [trampoline structure](https://www.datchley.name/recursion-tail-calls-and-trampolines/) to track the recursion in an alternate data structure or control structure to eliminate the risk of stack overflow

it should be noted that this work overlaps what's happening in #8083. if we get a fast internal PHP parser we might dramatically rewrite this function.

Reply:
@dmsnell Is it worth continuing with this considering you might change the PHP parser. The intention of this change is to fix rendering of inner blocks that doesn't happen at all right now.